src/openssl_wrap.cpp

```diff
@@ -1325,5 +1325,45 @@ std::vector<uint8_t> _EC_KEY_key2buf(const EVP_PKEY* evp, point_conversion_form_
     return result;
 }
 
+
+std::vector<uint8_t> _EVP_derive_key(const EVP_PKEY *peerkey, const EVP_PKEY *key)
+{
+    size_t secret_len;
+    /* EVP_PKEY_CTX_new: Why const_cast is allowed:
```

Comment:
The real question should be: Why is const_cast needed here and ok? Why do these contexts want mutable objects in the first place?

Reply:
Done